### PR TITLE
Update BOLT package

### PR DESCRIPTION
**Pull request by shintaro-iwasaki**
This PR adds two new versions (`1.0rc1` and `develop`) and missing dependencies (CMake and Argobots) to the BOLT package.
Note that this BOLT package previously uses a built-in Argobots, but it complicates the version control. The Argobots/BOLT team would like to separate it from the BOLT package.

### CODE REVIEW

File: var/spack/repos/builtin/packages/bolt/package.py [hunk 1]
```diff
@@ -19,17 +19,23 @@ class Bolt(CMakePackage):
 
     homepage = "http://www.bolt-omp.org/"
     url      = "https://github.com/pmodels/bolt/releases/download/v1.0b1/bolt-1.0b1.tar.gz"
+    git      = "https://github.com/pmodels/bolt.git"
 
+    version("develop", branch="master")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/bolt/package.py]: You can actually call this version `master` now that #1983 has been merged.
```suggestion
    version("master", branch="master")
```

File: var/spack/repos/builtin/packages/bolt/package.py [hunk 2]
```diff
@@ -19,17 +19,23 @@ class Bolt(CMakePackage):
...
     depends_on('automake', type='build')
+    depends_on('cmake', type='build')
     depends_on('libtool', type='build')
 
     def cmake_args(self):
         options = [
             '-DLIBOMP_USE_ITT_NOTIFY=off',
-            '-DLIBOMP_USE_ARGOBOTS=on'
+            '-DLIBOMP_USE_ARGOBOTS=on',
+            '-DLIBOMP_ARGOBOTS_INSTALL_DIR="' + self.spec['argobots'].prefix + '"'
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/bolt/package.py]: Quotes aren't needed here, Spack runs everything through Python, not through a shell.
```suggestion
            '-DLIBOMP_ARGOBOTS_INSTALL_DIR=' + self.spec['argobots'].prefix
```

File: var/spack/repos/builtin/packages/bolt/package.py [hunk 3]
```diff
@@ -19,17 +19,23 @@ class Bolt(CMakePackage):
...
+    git      = "https://github.com/pmodels/bolt.git"
 
+    version("develop", branch="master")
+    version("1.0rc1", "77733ba2ad9440c29b36d1c1411a9793")
     version("1.0b1", "df76beb3a7f13ae2dcaf9ab099eea87b")
 
     depends_on('autoconf', type='build')
+    depends_on('argobots')
     depends_on('automake', type='build')
+    depends_on('cmake', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/bolt/package.py]: This line isn't actually needed, the `CMakePackage` base class already adds it automatically.
```suggestion
```

**Comment by shintaro-iwasaki** [comment]
Thank you for the comments. I force-pushed the new commits, which fixed the three issues you pointed out.

File: var/spack/repos/builtin/packages/bolt/package.py [hunk 4]
```diff
@@ -20,16 +20,21 @@ class Bolt(CMakePackage):
     homepage = "http://www.bolt-omp.org/"
     url      = "https://github.com/pmodels/bolt/releases/download/v1.0b1/bolt-1.0b1.tar.gz"
 
+    version("master", branch="master")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/bolt/package.py]: You'll need to add a `git` branch above, that's why the unit tests are failing.

**Comment by shintaro-iwasaki** [comment]
I somehow removed the `git` branch at force-push. The latest one should have fixed this issue.

**Pull request merged**